### ISSUE TITLE
Error if updating a remote file.

### ISSUE THREAD

**Issue by thaibault** [siom79/jdrivesync#25]
When jdrviesync has to update a local (newer and changed) file over a remote one, I get an exception.

My command:

```
java -jar jdrivesync-0.4.3-jar-with-dependencies.jar --log-file ~/.jdrivesync.log --checksum --local-dir "/data" --remote-dir "/" --ignore-file "/jdrivesync_ignore.txt" --up
```

Relevant logging output with traceback:

```
...
2018-03-28 18:44:13 INFORMATION jdrivesync.gdrive.GoogleDriveAdapter updateFile(): Updating file 1c7HV8BcRzwk6Zukq71epZ76Yb5ocM1DA (/data/subfolder/info.txt).
2018-03-28 18:44:24 WARNUNG jdrivesync.sync.Synchronization$1 visitDirectory(): Skipping file/directory '/data/subfolder' because an exception occurred: null
java.lang.IllegalArgumentException
	at com.google.api.client.repackaged.com.google.common.base.Preconditions.checkArgument(Preconditions.java:111)
	at com.google.api.client.util.Preconditions.checkArgument(Preconditions.java:37)
	at com.google.api.client.googleapis.media.MediaHttpUploader.upload(MediaHttpUploader.java:331)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:427)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
	at jdrivesync.gdrive.GoogleDriveAdapter.lambda$updateFile$12(GoogleDriveAdapter.java:226)
	at jdrivesync.gdrive.RetryOperation.executeWithRetry(RetryOperation.java:28)
	at jdrivesync.gdrive.GoogleDriveAdapter.updateFile(GoogleDriveAdapter.java:226)
	at jdrivesync.sync.Synchronization$1.performChecksumCheck(Synchronization.java:264)
	at jdrivesync.sync.Synchronization$1.processRemoteChildFound(Synchronization.java:171)
	at jdrivesync.sync.Synchronization$1.visitDirectory(Synchronization.java:70)
	at jdrivesync.fs.FileSystemWalker.walkInternal(FileSystemWalker.java:58)
	at jdrivesync.fs.FileSystemWalker.walkInternal(FileSystemWalker.java:65)
	at jdrivesync.fs.FileSystemWalker.walkInternal(FileSystemWalker.java:65)
	at jdrivesync.fs.FileSystemWalker.walk(FileSystemWalker.java:34)
	at jdrivesync.sync.Synchronization.syncUp(Synchronization.java:52)
	at jdrivesync.App.sync(App.java:87)
	at jdrivesync.App.run(App.java:54)
	at jdrivesync.App.main(App.java:27)

2018-03-28 18:44:24 INFORMATION jdrivesync.report.NoopReport log(): Error(n.a.) Skipped /data/subfolder
...
```

The file "/data/subfolder/info.txt" exists locally and remotly but has updated locally.

When I remove the remote one, the file will be uploaded correctly.

My local folder "/data" lives on an encrypted "encfs" layer and I'm using ArchLinux with the following java version:

```
>>> java -version
openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-b12)
OpenJDK 64-Bit Server VM (build 25.162-b12, mixed mode)
```

Thanks for help, time and providing this great Tool!

**Comment by nukleos** [comment]
Unfortunately, I'm seeing exactly the same error message (running on Xubuntu 18.04)... and this makes it hard to recommend 'jdrivesync' for serious work. Any suggestions for an alternative solution?

**Comment by siom79** [comment]
Released with 0.5.0.
